inc/az_ulib_descriptor_api.h

```diff
@@ -0,0 +1,210 @@
+ *
+ * This structure contains the full information about a collection of action that represents the
+ * interface.
+ */
+typedef struct az_ulib_interface_descriptor_tag {
+  const char *name; /**<The `\0` terminated `const char*` with the interface name. */
+  az_ulib_version version; /**<The #az_ulib_version with the interface version. */
+  uint8_t size; /**<The `uint8_t` with the number of actions in the interface. */
+  az_ulib_action_descriptor *action_list; /**<The list of #az_ulib_action_descriptor with the
+                                          actions in this interface. */
+} az_ulib_interface_descriptor;
+
+/**
+ * @brief   Create a new Interface descriptor.
+ *
+ * This API creates a new description for a interface, it will create the
+ * #az_ulib_interface_descriptor with the provided interface name and version, adding all the
+ * actions.
+ *
+ * The descriptor will be stored in the Text area (Flash) that will be pointed by the provided
+ * interface_var. In this way, no other components on the system needs to copy any of the data on
+ * the descriptor.
+ *
+ * @param[in]   interface_var   The `az_ulib_interface_descriptor` that will point to the created
+ *                              descriptor.
+ * @param[in]   interface_name  The `/0` terminated `const char* const` with the interface name.
+ *                              It cannot be `NULL` and shall be allocated in a way that it stays
+ *                              valid until the interface is unpublished at some (potentially)
+ *                              unknown time in the future.
+ * @param[in]   version         The `az_ulib_version` with the interface version.
+ * @param[in]   ...             The list of `az_ulib_action_descriptor` with the actions in the
+ *                              interface.
+ */
+#define AZ_ULIB_DESCRIPTOR_CREATE(interface_var, interface_name, version, ...) \
+  static const az_ulib_action_descriptor MU_C2(interface_var, _ACTION_LIST)[] \
+      = { MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT, __VA_ARGS__) }; \
+  static const az_ulib_interface_descriptor MU_C1(interface_var) \
+      = { (interface_name), \
+          (version), \
+          (uint8_t)(MU_COUNT_ARG(__VA_ARGS__) / 4), \
+          (az_ulib_action_descriptor *)MU_C2(interface_var, _ACTION_LIST) };
+
+/**
+ * @brief   Add property to the interface descriptor.
+ *
+ * Populate a new `property` action descriptor to add to the interface. On the interface context, a
+ * `property` is a data exposed by its `get` and `set` function.
+ *
+ * @param[in]   name        The `/0` terminated `const char* const` with the property name. It
+ *                          cannot be `NULL` and shall be allocated in a way that it stays valid
+ *                          until the interface is unpublished at some (potentially) unknown time
+ *                          in the future.
+ * @param[in]   get         The function pointer to `az_ulib_action_get` with the implementation
```

Comment:
>az_ulib_action_get [](start = 53, length = 18)

these references to typedef'd functions can be preceded by either `#` (in which case it becomes `#az_ulib_action_get` or followed by `()` in which case it becomes `az_ulib_action_get()`. either one would work. i think i prefer in the case of typedef'd functions for it to look like a function (so following with `()`)